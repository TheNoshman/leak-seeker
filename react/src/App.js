import './css/app.css'
import Navbar from './components/navbar'
import OverallMainContainer from './containers/overallmaincontainer'

function App () {
  return (
    <div className='App'>
      <Navbar />
      <OverallMainContainer className='glass' />
      <div className='circle1' />
      <div className='circle2' />
      <div className='circle3' />
      <div className='circle4' />
    </div>
  )
}

export default App
