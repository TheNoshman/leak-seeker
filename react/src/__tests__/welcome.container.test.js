import React from 'react'
import App from '../App'
import { screen, render, fireEvent, cleanup } from '@testing-library/react'

describe('Welcome Page', () => {
  afterEach(() => {
    cleanup
  })

  it('should take a snapshot of the welcome screen', () => {
    const { asFragment } = render(<App />)

    expect(asFragment(<App />)).toMatchSnapshot()
  })

  it('should allow users to search by registration', () => {
    render(<App />)

    fireEvent.click(screen.getByText(/search faults/i))
    expect(screen.getByPlaceholderText(/enter a registration number.../i)).toBeTruthy()

    const inputForm = screen.getByPlaceholderText(/enter a registration number.../i)
    expect(inputForm.value).toBe('')
    fireEvent.change(inputForm, { target: { value: 'ABC123'} })
    expect(inputForm.value).toBe('ABC123')

    const submitBtn = screen.getByRole('button', { name: /submit/i })
    expect(submitBtn).toBeTruthy()
    fireEvent.click(submitBtn)
  })
})