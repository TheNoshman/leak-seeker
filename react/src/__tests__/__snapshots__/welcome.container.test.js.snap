// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Page should take a snapshot of the welcome screen 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="navbar-cont glass"
    >
      <section
        class="navbar-left"
      >
        <img
          alt="brand logo"
          class="navbar-logo-img"
          src="logo.png"
        />
        <h1>
          LeakSeeker
        </h1>
      </section>
      <section
        class="navbar-right"
      >
        <a
          class="navbar-right-item"
          href="x"
        >
          Home
        </a>
        <a
          class="navbar-right-item"
          href="x"
        >
          About
        </a>
        <a
          class="navbar-right-item"
          href="x"
        >
          Contact
        </a>
        <button
          class="navbar-right-item login-btn"
          type="button"
        >
          Login
        </button>
      </section>
    </div>
    <div
      class="mainpage"
    >
      <div
        class="welcome-container glass"
      >
        <h1>
          Welcome to Leak Seeker
        </h1>
        <h3>
          Please choose an option:
        </h3>
        <div
          class="button-container"
        >
          <div
            class="button-div"
          >
            <a
              class="welcome-choice-btn"
              href="#"
            >
              Search Faults
            </a>
          </div>
          <div>
            <img
              alt="brand logo"
              class="welcome-logo"
              src="logo.png"
            />
          </div>
          <div
            class="button-div"
          >
            <a
              class="welcome-choice-btn"
              href="#"
            >
              Register Fault
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="circle1"
    />
    <div
      class="circle2"
    />
    <div
      class="circle3"
    />
    <div
      class="circle4"
    />
  </div>
</DocumentFragment>
`;
